parser/src/error.rs

```diff
@@ -143,10 +147,19 @@ impl From<LalrpopError<Location, Tok, LexicalError>> for ParseError {
                 error: ParseErrorType::Lexical(error.error),
                 location: error.location,
             },
-            LalrpopError::UnrecognizedToken { token, expected } => ParseError {
-                error: ParseErrorType::UnrecognizedToken(token.1, expected),
-                location: token.0,
-            },
+            LalrpopError::UnrecognizedToken { token, expected } => {
+                // Hacky, but it's how CPython does it. See PyParser_AddToken,
+                // in particular "Only one possible expected token" comment.
+                let expected = if expected.len() == 1 {
```

Comment:
This might be an opportunity to use the [`first`](https://doc.rust-lang.org/std/primitive.slice.html#method.first) function.

Reply:
No, because the intent is to get `None` if `expected` has more than two elements.

Reply:
Ah, okay, I understand. Thank you!